ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -394,14 +409,22 @@ void RSkComponentImage::requestNetworkImageData(ImageSource source) {
 
     // TODO : Parse request headers and retrieve caching details
 
-    cacheExpiryTime_ = std::min(responseMaxAgeTime,static_cast<double>(DEFAULT_MAX_CACHE_EXPIRY_TIME));
+    cacheExpiryTime_ = std::min(MILLSEC_CONVERTER(responseMaxAgeTime),static_cast<double>(DEFAULT_MAX_CACHE_EXPIRY_TIME));
     RNS_LOG_DEBUG("url [" << responseData->responseurl << "] canCacheData[" << canCacheData_ << "] cacheExpiryTime[" << cacheExpiryTime_ << "]");
     return 0;
   };
 
 
   // completioncallback lambda fuction
-  auto completionCallback =  [this, remoteCurlRequest](void* curlresponseData,void *userdata)->bool {
+  auto completionCallback =  [this, weakThis = this->weak_from_this(), remoteCurlRequest](void* curlresponseData,void *userdata)->bool {
+    auto isAlive = weakThis.lock();
+    if(!isAlive) {
+      RNS_LOG_WARN("This object is already destroyed. ignoring the completion callback");
+      return 0;
+    }
+    networkRequestMutext.lock();
+    isRequestinProgress_=false;
```

Comment:
will remoteCUrlRequest be destroyed in this case or should we reset the headercallback and completion callback handles ?